config/config-builder/src/validator_config.rs

```diff
@@ -160,27 +204,12 @@ impl ValidatorConfig {
         validator_network.advertised_address = self.advertised.clone();
         validator_network.seed_peers = seed_peers;
 
+        self.build_key_manager(&mut config)?;
         self.build_safety_rules(&mut config)?;
 
         Ok(config)
     }
 
-    pub fn build_set(&self) -> Result<Vec<NodeConfig>> {
-        let (configs, _) = self.build_common(false)?;
-        Ok(configs)
-    }
-
-    pub fn build_faucet_client(&self) -> Result<(Ed25519PrivateKey, Waypoint)> {
```

Comment:
for example, this is only here, because validators / faucet are core to the system, but we could have had a faucet config and the validator config depended on the faucet config.